### PR TITLE
Refactor `espressif` CI

### PR DESCRIPTION
**Pull request by MicroDev1**
~~This will install toolchain specific to the chip on an espressif board.~~
See https://github.com/adafruit/circuitpython/pull/7420#issuecomment-1372643074.

### CODE REVIEW

**Comment by jepler** [comment]
how does this interact with the idf tools cache?  For instance, if the cache is created from an xtensa build, will it fail to cache the riscv toolchain, thus causing all riscv toolchain builds to have to fetch from the remote network instead of from cache?

**Comment by MicroDev1** [comment]
I completely forgot about the cache, I am not sure how this change will interact with it and I won't give it much thought as the cache makes this change pointless as there is no performance improvement.

I can alter this PR and just keep the slight refactoring stuff or this can be closed.

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -412,14 +412,14 @@ jobs:
...
     - name: Clone IDF submodules
-      run: |
-        (cd $IDF_PATH && git submodule update --init)
+      run: git submodule update --init $IDF_PATH
       env:
         IDF_PATH: ${{ github.workspace }}/ports/espressif/esp-idf
     - name: Install IDF tools
       run: |
-        $IDF_PATH/tools/idf_tools.py --non-interactive install required
-        $IDF_PATH/tools/idf_tools.py --non-interactive install cmake
```
Comment (jepler) [inline .github/workflows/build.yml:422]: From what I read, "install required" is equivalent to bare "install".  so the effect of this is to skip installing cmake.  This has the effect of changing the version of cmake used. Before this, the version from esp-idf is used.  after this, a version installed by github is used.  Is it important to use a matching version of cmake, or will any version do?

**Review by jepler — COMMENTED** [review]
Thanks, but I have a question.

File: .github/workflows/build.yml (hunk 1 above)
Comment (MicroDev1) [reply .github/workflows/build.yml:422]: I replicated the `install.sh` script here which doesn't seem to install `cmake` but I think it is better to have the idf specific `cmake` so I'll revert this. Thanks for pointing it out.

**Comment by MicroDev1** [comment]
Changes moved to PR 7428.

**Pull request closed without merging**